backend/de.metas.util.web/src/main/java/de/metas/util/web/audit/dto/ApiRequest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * #%L
+ * de.metas.util.web
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.util.web.audit.dto;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import org.springframework.util.LinkedMultiValueMap;
+
+import javax.annotation.Nullable;
+
+@Value
+@Builder
+public class ApiRequest
+{
+	@NonNull
+	LinkedMultiValueMap<String, String> headers;
```

Comment:
any reason or not using ImmutableMultimap or ImmutableMultiset which is immutable?

Reply:
please change to an immutable variant or discuss why it's not possible

Reply:
reiterated the request